lib/ws_servers/api/handlers/on_order_submit.js

```diff
@@ -17,35 +17,44 @@ module.exports = async (server, ws, msg) => {
     authToken: { type: 'string', v: authToken },
     packet: { type: 'object', v: packet }
   })
+  const sendStatus = (status) => send(ws, ['data.order.submit_status', status, packet ? packet.type : null])
 
   if (!validRequest) {
+    sendStatus('failed')
     return
   }
 
   if (!isAuthorized(ws, authToken)) {
+    sendStatus('failed')
     return sendError(ws, 'Unauthorized')
   } else if (exID !== 'bitfinex') {
+    sendStatus('failed')
     return sendError(ws, 'Unrecognised exchange, cannot submit order')
   } else if (!ws.clients[exID]) {
+    sendStatus('failed')
     return sendError(ws, `No client open for ${_capitalize(exID)}`)
+  } else if (exID != 'bitfinex') {
```

Comment:
repeated else if condition

Reply:
indeed, it was present in the 'default' option of 'switch' statement, but it turns out that it was already dead code